.github/workflows/rspec-tests.yml

```diff
@@ -0,0 +1,22 @@
+name: RSpec Tests
+run-name: ${{ github.actor }} is running rspec tests
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  run-specs:
+    runs-on: macos-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Set up ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.1'
```

Comment:
Does the version need to be hardcoded here?

Reply:
no, if there's a `.ruby_version` file, it will read it from there. I hardcoded it for now just for testing.

Reply:
updated it to use `.rub-version` file